### ISSUE TITLE
Facebook login fails to parse JSON properly

### ISSUE THREAD

**Issue by thekoding** [FusionAuth/fusionauth-issues#648]
### Description 
When I try to login with Facebook by calling the following PHP code, I get the following error:
```php
$loginResponse = $fusionAuthClient->identityProviderLogin([
            'applicationId' => "MY_APP_ID",
            'identityProviderId' => "MY_FACEBOOK_IDENTITY_PROVIDER_ID",
            'data' => [
                'token' => $data->getAccessToken()
            ],
            'metaData' => [
                'device' => [
                    'type' => $data->getDeviceType()
                ]
            ]
 ]);
```
And the error I get in the event logs is the following:

```
Request to the [https://graph.facebook.com/v3.1/me] endpoint failed. Status code [200]

Exception encountered.

com.inversoft.rest.JSONException : Message: Failed to parse the HTTP response as JSON. Actual HTTP response body:
{"email":"REDACTED","picture":{"data":{"height":50,"is_silhouette":false,"url":"REDACTED","width":50}},"id":"REDACTED"}
```
### Steps to reproduce
1. Configure Facebook identity provider and enable debug mode.
2. Use the identity provider endpoint, passing a valid Facebook identity provider id and access token
3. Check event logs for error

### Expected behavior
I expected for the login to work properly and to get an access token and a refresh token.

### Platform
(Please complete the following information)
 - Server: Symfony
 - OS: Linux, macOS
 - Browser + version:  Postman 7.24.0

**Comment by robotdan** [comment]
1. What version of FusionAuth?
2. What permissions and fields have you configured with the Facebook IdP?
3. Can you provide the actual JSON response from the event log (sanitized if necessary) - but it would be helpful to see if based upon the actual JSON if it should be parseable.

**Comment by thekoding** [comment]
1. FusionAuth reports 1.15.8 on the left, even though I have just upgraded to 1.16.1
2. 
- fields: email,picture
- permissions: email,public_profile

3. I have uploaded the json below (compressed because github does not support uploading .json files (as I'd like to avoid posting my facebook details directly into an issue if I could).
[res.json.zip](https://github.com/FusionAuth/fusionauth-issues/files/4684116/res.json.zip)

**Comment by robotdan** [comment]
Thanks, that helped. I was able to recreate your symptom. 

Try removing `picture` from the fields. 

I think the issue is that we are currently making a second request the `https://graph.facebook.com/v3.1/me/picture` API to collect the picture. 

When you request `picture` - that value is coming back on the `Me` API which we are not expecting. 

Because we are making this request anyway - you can safely omit the `picture` field from your configuration and we'll still collect the image URL. 

We should be able to get this fixed in an upcoming release so that we can handle the `picture` claim and pick it up on the first response so we can skip the second API call.

**Comment by thekoding** [comment]
Thanks, it worked after I removed the picture from fields. Should I close the issue now, since it's been added to the backlog already?

**Comment by robotdan** [comment]
You can leave it open, I'll use this to track the issue to completion. It will be fixed in version 1.17.0.

**Comment by Interchange5** [comment]
@robotdan https://rock-complementary2.github.io/MB1Mz8BCqhL1es2DHj6Ru/
